content/client-lib-development-guide/features.textile

```diff
@@ -84,6 +84,11 @@ h3(#restclient). RestClient
 * @(RSC8)@ Supports two protocols:
 ** @(RSC8a)@ "MessagePack":http://msgpack.org/ binary protocol (this is the default for environments having a suitable level or support for binary data)
 ** @(RSC8b)@ JSON text protocol (used when @useBinaryProtocol@ option is false)
+** @(RSC8c)@ The library sets the @Accept@ and @Content-Type@ type headers, to reflect whether the client is configured to use a binary or JSON based protocol, to @application/x-msgpack@ or @application/json@ respectively.
+** @(RSC8d)@ Irrespective of the specified @Accept@ header, the library should attempt where possible to process any response body based on the content type indicated in a @Content-Type@ header.
+** @(RSC8e)@ If the library is unable to process the specified content type, then the library processes the response as follows.
+*** @(RSC8e1)@ In the case of a @statusCode@ indicating an error response, the error must be propagated back to the caller with as much information as it is possible to obtain from the response, including the @statusCode@ and, where applicable, an error message containing text from the response body to help to diagnose the cause of the error.
```

Comment:
This is a little vague, isn't it? What code should I write to make my client library compliant? Is there a limit? What if I receive some Textile from the server, for example? Am I meant to have a parser for that? **Needs scope.**

Reply:
refined, tighter spec. https://github.com/ably/docs/pull/888/commits/bfcb292156fecf01102086c64cd951f7b018c8b5 

@paddybyers, presumably the `base64-encoding` has been specified in case the response body is not textual - in which case we expect to see base64 in the debug messages and we then decode that as part of our debugging workflow?

Reply:
I've suggested base64 unconditionally, yes, because otherwise you'd be having to build knowledge into the library about which content types are binary and which aren't. Nothing is going to inspect these encoded messages other than humans when investigating issues.